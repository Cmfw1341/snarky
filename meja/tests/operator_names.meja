let (+) = 15;

let (-) = 20;

let (!) = fun (_) => {30;};

let (~-) = fun (_) => {80;};

let (||) = fun (x, y) => {
  switch (x) {
    | true => x
    | false => y
  };
};

let a = (||) (true, false);

let b = true || false;

let c = !a;

let d = - 35;

let e = - d;

let f : (int -> int -> int) -> int = fun (check : 'a -> 'a -> 'a) => {
  let (+) = fun (l, _) => {l;};
  let (-) = fun (_, r) => {r;};
  let (*) = fun (_, r) => {r;};
  let (/) = fun (l, _) => {l;};
  let x = () * true - 1i / ((), ()) + (1i, 1i);
  let y = (() * true) - (1i / ((), ())) + (1i, 1i);
  check (x, y);
};
