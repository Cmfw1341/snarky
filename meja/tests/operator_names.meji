let (+) : int;  let (-) : int;  let (!) : _ -> int;  let (~-) : _ -> int; 
let (||) : bool -> bool -> bool;  let a : bool;  let b : bool;  let c : int; 
let d : int;  let e : int;  let f : (int -> int -> int) -> int; 

