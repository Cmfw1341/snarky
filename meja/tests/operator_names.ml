module Impl = Snarky.Snark.Make (Snarky.Backends.Mnt4.Default)
open Impl

let ( + ) = Field.constant (Field.Constant.of_string "15")

let ( - ) = Field.constant (Field.Constant.of_string "20")

let ( ! ) _ = Field.constant (Field.Constant.of_string "30")

let ( ~- ) _ = Field.constant (Field.Constant.of_string "80")

let ( || ) x y = match x with true -> x | false -> y

let a = true || false

let b = true || false

let c = !a

let d = -Field.constant (Field.Constant.of_string "35")

let e = -d

let (f : (int -> int -> int) -> int) =
 fun (check : 'a -> 'a -> 'a) ->
  let ( + ) l _ = l in
  let ( - ) _ r = r in
  let ( * ) _ r = r in
  let ( / ) l _ = l in
  let x = (((() * true) - 1) / ((), ())) + (1, 1) in
  let y = (() * true) - (1 / ((), ())) + (1, 1) in
  check x y
