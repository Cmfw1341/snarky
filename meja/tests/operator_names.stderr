File "tests/operator_names.meja", line 11, characters 6-10:
Error: Unbound constructor true.
